build error on ghc 9.2.3 with stack

On MacOS with M1 / aarch64-osx. Error occurs when installing this package with stack `nightly-2022-06-22`, using the following `extra-deps` (suggested by stack):

```
- beam-automigrate-0.1.2.0@sha256:92f1117ba8a4bcb174f7d061358486e0a09e43ba9d80f965477e71edd2df6704,6249
- gargoyle-postgresql-connect-0.1.0.0@sha256:b33e13dcaea7e2387241bbe57f4ec67b8715ee4e44b68e7ddb3c35dd5da74f14,1242
- gargoyle-0.1.1.0@sha256:c8fafaf0da5be7d222945fa39e10b04d8cc9b49b02b869221eb857ea04c2bd76,1629
- gargoyle-postgresql-0.2.0.1@sha256:776b40cca3a9d4222212baecb364da203e769d5046dc3079aa05749164448097,3679
- gargoyle-postgresql-nix-0.3.0.0@sha256:40bae12ca9985c88b5d0969f1cb07ec7b83f85b2797fe0982e8cb839b85a9106,2066
- posix-escape-0.1@sha256:adbb650397cd8685ea220b40739ee80d4e8329643a98aa2993a6f13a598bd18c,1160
- which-0.2.0.1@sha256:807c087f3cd345133fb79f7fb54e630cbff210f74018f1dd219d497076221108,907
```

The build fails with the following error:

```
beam-automigrate                 >       except perhaps to import instances from ‘Database.Beam.Schema’
beam-automigrate                 >     To import instances alone, use: import Database.Beam.Schema()
beam-automigrate                 >    |
beam-automigrate                 > 16 | import Database.Beam.Schema (Beamable, PrimaryKey, TableEntity, TableSettings)
beam-automigrate                 >    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
beam-automigrate                 > [ 5 of 11] Compiling Database.Beam.AutoMigrate.Compat
beam-automigrate                 >
beam-automigrate                 > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-7a4f5bd52d1b3196/beam-automigrate-0.1.2.0/src/Database/Beam/AutoMigrate/Compat.hs:86:3: error:
beam-automigrate                 >     • Could not deduce (Ord (SchemaConstraint a))
beam-automigrate                 >         arising from the superclasses of an instance declaration
beam-automigrate                 >       from the context: (IsMaybe a ~ nullary,
beam-automigrate                 >                          HasSchemaConstraints' nullary a)
beam-automigrate                 >         bound by the instance declaration
beam-automigrate                 >         at src/Database/Beam/AutoMigrate/Compat.hs:(86,3)-(89,24)
beam-automigrate                 >     • In the instance declaration for ‘HasSchemaConstraints a’
beam-automigrate                 >    |
beam-automigrate                 > 86 |   ( IsMaybe a ~ nullary,
beam-automigrate                 >    |   ^^^^^^^^^^^^^^^^^^^^^^...
```

--- a/src/Database/Beam/AutoMigrate/Compat.hs
+++ b/src/Database/Beam/AutoMigrate/Compat.hs
@@ -86,7 +86,8 @@ instance HasSchemaConstraints' 'False (Beam.TableField e t) where
 
 instance
   ( IsMaybe a ~ nullary,
-    HasSchemaConstraints' nullary a
+    HasSchemaConstraints' nullary a,
+    Ord (SchemaConstraint a)
   ) =>
   HasSchemaConstraints a
   where

